android/test-app/src/main/java/com/arthenica/mobileffmpeg/test/SafTabFragment.java

```diff
@@ -0,0 +1,426 @@
+        final File image3File = new File(requireContext().getCacheDir(), "tajmahal.jpg");
+        final String videoPath = Config.getSafParameterForReadAndWrite(requireContext(), outUri);
+
+        try {
+            // IF VIDEO IS PLAYING STOP PLAYBACK
+            videoView.stopPlayback();
+
+            videoView.setVisibility(View.GONE);
+            outputText.setVisibility(View.VISIBLE);
+
+            String selectedCodec = getCodec(videoPath);
+            Log.d(TAG, String.format("Testing VIDEO encoding with '%s' codec", selectedCodec));
+
+            showProgressDialog();
+
+            ResourcesUtil.resourceToFile(getResources(), R.drawable.colosseum, image1File);
+            ResourcesUtil.resourceToFile(getResources(), R.drawable.pyramid, image2File);
+            ResourcesUtil.resourceToFile(getResources(), R.drawable.tajmahal, image3File);
+
+            final String ffmpegCommand = Video.generateEncodeVideoScript(image1File.getAbsolutePath(), image2File.getAbsolutePath(), image3File.getAbsolutePath(), videoPath, selectedCodec, getCustomOptions(selectedCodec));
+
+            Log.d(TAG, String.format("FFmpeg process started with arguments\n'%s'.", ffmpegCommand));
+
+            long executionId = FFmpeg.executeAsync(ffmpegCommand, new ExecuteCallback() {
+
+                @Override
+                public void apply(final long executionId, final int returnCode) {
+                    Log.d(TAG, String.format("FFmpeg process exited with rc %d.", returnCode));
+
+                    Log.d(TAG, "FFmpeg process output:");
+
+                    Config.printLastCommandOutput(Log.INFO);
+
+                    hideProgressDialog();
+
+                    MainActivity.addUIAction(new Callable<Object>() {
+
+                        @Override
+                        public Object call() {
+                            if (returnCode == RETURN_CODE_SUCCESS) {
+                                Log.d(TAG, "Encode completed successfully; playing video.");
+                                playVideo();
+                            } else {
+                                Popup.show(requireContext(), "Encode failed. Please check log for the details.");
+                                Log.d(TAG, String.format("Encode failed with rc=%d.", returnCode));
+                            }
+
+                            return null;
+                        }
+                    });
+                }
+            });
+
+            Log.d(TAG, String.format("Async FFmpeg process started with executionId %d.", executionId));
+
+        } catch (IOException e) {
+            Log.e(TAG, String.format("Encode video failed %s.", Exceptions.getStackTraceString(e)));
+            Popup.show(requireContext(), "Encode video failed");
+        }
+    }
+
+    private void playVideo() {
+        videoView.setVisibility(View.VISIBLE);
+        outputText.setVisibility(View.GONE);
```

Comment:
Could you please simplify this section without hiding a component? It is not useful.

I clicked the `RUN FFMPEG` button. Text area included the encoding logs during the execution but I couldn't follow them because of the progress dialog. When execution ended, video started to play. Still couldn't browse encoding logs. Video playing ended and this time `ffprobe` was executed. So, it is not possible to follow encoding logs and see what's happening.

Reply:
I believe that video playback is nice here, but I have removed the `ffprobe` part.

Reply:
I agree, it is nice. I just couldn't get used to the idea of hiding and showing components. This is not something I like to do. Anyway, it can stay if you're happy with it.

Reply:
Both video playback and ffmpeg log contribute to showing off this use case. You are right, the ffprobe step was too much.